app/classes/observation_report/darwin/images.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+module ObservationReport
+  module Darwin
+    # Darwin Core Observations format.
+    class Images < ObservationReport::CSV
+      attr_accessor :observations, :query, :tables
+
+      self.separator = "\t"
+
+      def initialize(args)
+        super(args)
+        self.observations = args[:observations]
+        initialize_query
+      end
+
+      def initialize_query
+        add_joins
+        add_project
+      end
+
+      def add_joins
+        self.tables = {
+          images: Image.arel_table,
+          images_observations: Arel::Table.new(:images_observations),
+          observations: Observation.arel_table,
+          users: User.arel_table,
+          licenses: License.arel_table
+        }
```

Comment:
These arel_tables are pretty slick!  I think I've finally gotten my head around what's going on here.

Question: Did you consider making self.tables a method instead?  It's a perfect opportunity to use that screwy-named technique... that's it: memoization! :)

def tables
  @tables ||= {
    images: Image.arel_table,
    etc.
  }
end

Really just serves to remove this assignment to self.tables from add_joins, but that seems worthwhile.  Initializing the arel_tables seems to be sort of foundational to this style of building a query, so burying that step inside "add_joins" seems a bit strange to me.  I'm just saying, I guess, that it would have saved me a great deal of head-scratching if the arel_tables were instead explicitly "initialized" or at least have them implicitly spring into existence (via a memoized accessor).  Then I would have known that they were not just something needed by the joins.  For example, I was mightily confused when you seemingly arbitrarily assigned tables[:images] to self.query!

Formal suggestion:

1. move the line self.query = tables(:images) to the top of initialize_query
2. move self.tables = { ... } into a memoized method "tables"

I think the flow and structure would be a little bit clearer.  (To me at least!)

Reply:
This is the first time I've used the Arel stuff.  I kind of knew about it in the back of my head and it definitely took some head scratching to figure it out, but I ended up liking it pretty well.  Happy to clean it up some now that it's working.

Ha, I thought about going the memoization route, but I thought you didn't like it and I couldn't remember how it works off the top of my head.  Happy to do it that way.